checkpoint_ppo/3090_1223_FeatureCombine_Turning/config_info.txt
num_agents: 30
agent_radius: 0.5
num_rays: 50
max_ray_distance: 10.0
ray_vision_lamda: 0.2
max_speed_orientation: 1.6
max_speed_perpendicular: 0.5
max_acceleration: 2.0
max_angular_velocity: 0.7853981633974483
agent_terrain_obs_range: [15.0, 15.0]
start_env_size: [40.0, 40.0]
end_env_size: [40.0, 40.0]
env_scale_frequency: 20
dt: 0.04
action_interval_step_num: 1
min_walking_distance: 10.0
max_alive_time: 100.0
max_time_overlapped_with_obstacle: 2.0
agent_terrain_threshold: 0.9
implicit_agent_spawn_threshold: 0.0
implicit_agent_target_threshold: 0.9
update_nav_point_threshold: 5.0
auto_launch_ue: True
ip: 127.0.0.1
port: 12345
ue_command: C:\Program Files\Epic Games\UE_5.1\Engine\Binaries\Win64\UnrealEditor.exe
uproject_path: C:\Users\50191\Desktop\RLInteractiveCrowd-cpp\RLInteractiveCrowd_UE\RLInteractiveCrowd.uproject
map_path: /Game/Maps/L_RLTrain.umap
data_buffer_size: 200000
num_user_agents_chase: 0
num_user_agents_random: 0
num_obstacles: 0
terrain_filename: terrain_grayscale_256x256.png
terrain_obs_dim: [32, 32]
alive_penalty: 7.0
velocity_smooth_penalty: 0.25
collision_penalty: 100.0
target_approach_distance_reward: 100.0
target_away_distance_penalty_ratio: 1.0
use_nav_point: False
nav_point_approach_distance_reward: 0.0
target_arrive_reward: 1000.0
obstacle_penalty: 1.0
obstacle_time_coef: 0.1
terrain_penalty: 50.0
train_strategy: single_individual_feature_combine_weight_net
model_name: ppo
ablation_type: RNT
train_terrain_specific_name: turning
terrain_prev_set: []
terrain_prev_prob: 0.2
n_episodes: 10000
num_new_agents_per_episode: 200
lr_actor: 2.5e-05
lr_critic: 0.00025
fc1_dims: 400
fc2_dims: 300
fc2_dim_img: 150
pretrained_model_path: 
pretrained_entity_model_path: checkpoint_ppo/3090_1218_WeightMLP_Pretrained
pretrained_implicit_model_path: checkpoint_ppo/3090_1218_WeightMLP_Pretrained
fix_entity: False
use_user_weight_range: False
entity_weight_range: [1.0, 1.0]
implicit_weight_range: [1.0, 1.0]
delay_learn_step: 256
general_update_frequency: 20
update_best_avg_num: 100
checkpoint_save_freq: 50
action_std_decay_freq: 5000
ppo_max_mem: 256
action_std_init: 0.6
action_std_decay_rate: 0.005
min_action_std: 0.6
ppo_batch_size: 32
ppo_n_epochs: 8
ddpg_max_mem: 100000
ddpg_batch_size: 1024
td3_max_mem: 100000
td3_batch_size: 64
policy_noise: 0.1
noise_clip: 0.25
explore_noise: 0.1
eval_n_episodes: 100
load_time_str: 
eval_env_size: [40.0, 40.0]
eval_save_frame: True
eval_terrain_specific_name: white
eval_train_type: implicit
eval_reward_weight_collision: 1.0
eval_reward_weight_terrain: 1.0
log_update_freq: 100
device: cuda:0


config.py
import argparse
import ast
import torch
import os
import numpy as np


def parse_args():
    parser = argparse.ArgumentParser(description='RLInteractiveCrowd')
    # Agent config
    parser.add_argument('--num_agents', type=int, default=20)
    parser.add_argument('--agent_radius', type=float, default=0.5)
    parser.add_argument('--num_rays', type=int, default=50)
    parser.add_argument('--max_ray_distance', type=float, default=10.0)
    parser.add_argument('--ray_vision_lamda', type=float, default=0.2)
    parser.add_argument('--max_speed_orientation', type=float, default=1.6)
    parser.add_argument('--max_speed_perpendicular', type=float, default=0.5)
    parser.add_argument('--max_acceleration', type=float, default=2.0)
    parser.add_argument('--max_angular_velocity', type=float, default=0.25 * np.pi)
    parser.add_argument('--agent_terrain_obs_range', type=ast.literal_eval, default=[15.0, 15.0])

    # Env config
    parser.add_argument('--start_env_size', type=ast.literal_eval, default=[40.0, 40.0])
    parser.add_argument('--end_env_size', type=ast.literal_eval, default=[40.0, 40.0])
    parser.add_argument('--env_scale_frequency', type=int, default=20)
    parser.add_argument('--dt', type=float, default=0.04)
    parser.add_argument('--action_interval_step_num', type=int, default=1)
    parser.add_argument('--min_walking_distance', type=float, default=10.0)
    parser.add_argument('--max_alive_time', type=float, default=100.0)
    parser.add_argument('--arrive_range_scale', type=float, default=4.0)
    parser.add_argument('--target_loc_margin', type=float, default=3.0)
    parser.add_argument('--max_time_overlapped_with_obstacle', type=float, default=2.0)
    parser.add_argument('--agent_terrain_threshold', type=float, default=0.9)
    parser.add_argument('--implicit_agent_spawn_threshold', type=float, default=0.0)
    parser.add_argument('--implicit_agent_target_threshold', type=float, default=0.9)
    parser.add_argument('--update_nav_point_threshold', type=float, default=5.0)
    parser.add_argument('--auto_launch_ue', type=bool, default=True)
    parser.add_argument('--ip', type=str, default='127.0.0.1')
    parser.add_argument('--port', type=int, default=12345)
    parser.add_argument('--ue_command', type=str, default=r"C:\Program Files\Epic Games\UE_5.1"
                                                          r"\Engine\Binaries\Win64\UnrealEditor.exe")
    parser.add_argument('--uproject_path', type=str, default=os.getcwd() +
                                                             r"\RLInteractiveCrowd_UE\RLInteractiveCrowd.uproject")
    parser.add_argument('--map_path', type=str, default=r"/Game/RLCrowd/Maps/L_RLTrain.umap")
    parser.add_argument('--data_buffer_size', type=int, default=200000)
    parser.add_argument('--num_user_agents_chase', type=int, default=0)
    parser.add_argument('--num_user_agents_random', type=int, default=0)
    parser.add_argument('--num_obstacles', type=int, default=0)
    parser.add_argument('--terrain_filename', type=str, default='route_1.png')
    parser.add_argument('--terrain_obs_dim', type=ast.literal_eval, default=[32, 32])

    # Reward config
    parser.add_argument('--alive_penalty', type=float, default=7.0)
    parser.add_argument('--velocity_smooth_penalty', type=float, default=0.25)
    parser.add_argument('--collision_penalty', type=float, default=100.0)
    parser.add_argument('--target_approach_distance_reward', type=float, default=100.0)
    parser.add_argument('--target_away_distance_penalty_ratio', type=float, default=1.0)
    parser.add_argument('--use_nav_point', action='store_true', help='Enable navigation point')
    parser.add_argument('--no_use_nav_point', action='store_false', dest='use_nav_point',
                        help='Disable navigation point')
    parser.add_argument('--nav_point_approach_distance_reward', type=float, default=0.0)
    parser.add_argument('--target_arrive_reward', type=float, default=1000.0)
    parser.add_argument('--obstacle_penalty', type=float, default=1.0)
    parser.add_argument('--obstacle_time_coef', type=float, default=0.1)
    parser.add_argument('--terrain_penalty', type=float, default=50.0)

    # Training config
    # General
    parser.add_argument('--train_strategy', type=str, default='single_individual')
    parser.add_argument('--model_name', type=str, default='ppo')
    parser.add_argument('--ablation_type', type=str, default='RNT')
    parser.add_argument('--train_terrain_specific_name', type=str, default='white')
    parser.add_argument('--terrain_prev_set', type=ast.literal_eval, default=[])
    parser.add_argument('--terrain_prev_prob', type=float, default=0.2)
    parser.add_argument('--n_episodes', type=int, default=5000)
    parser.add_argument('--num_new_agents_per_episode', type=int, default=200)
    parser.add_argument('--lr_actor', type=float, default=0.000025)
    parser.add_argument('--lr_critic', type=float, default=0.00025)
    parser.add_argument('--fc1_dims', type=int, default=400)
    parser.add_argument('--fc2_dims', type=int, default=300)
    parser.add_argument('--fc2_dim_img', type=int, default=150)
    parser.add_argument('--pretrained_model_path', type=str, default='')
    parser.add_argument('--pretrained_entity_model_path', type=str, default='')
    parser.add_argument('--pretrained_implicit_model_path', type=str, default='')
    parser.add_argument('--fix_entity', action='store_true')
    parser.add_argument('--no_fix_entity', action='store_false', dest='fix_entity')
    parser.add_argument('--use_user_weight_range', action='store_true')
    parser.add_argument('--no_use_user_weight_range', action='store_false', dest='use_user_weight_range')
    parser.add_argument('--entity_weight_range', type=ast.literal_eval, default=[1.0, 1.0])
    parser.add_argument('--implicit_weight_range', type=ast.literal_eval, default=[1.0, 1.0])
    parser.add_argument('--delay_learn_step', type=int, default=256)
    parser.add_argument('--general_update_frequency', type=int, default=20)
    parser.add_argument('--update_best_avg_num', type=int, default=100)
    parser.add_argument('--checkpoint_save_freq', type=int, default=50)
    # PPO
    parser.add_argument('--action_std_decay_freq', type=int, default=50000)
    parser.add_argument('--ppo_max_mem', type=int, default=256)
    parser.add_argument('--action_std_init', type=float, default=0.6)
    parser.add_argument('--action_std_decay_rate', type=float, default=0.0005)
    parser.add_argument('--min_action_std', type=float, default=0.6)
    parser.add_argument('--ppo_batch_size', type=int, default=32)
    parser.add_argument('--ppo_n_epochs', type=int, default=8)
    # DDPG
    parser.add_argument('--ddpg_max_mem', type=int, default=100000)
    parser.add_argument('--ddpg_batch_size', type=int, default=1024)
    # TD3
    parser.add_argument('--td3_max_mem', type=int, default=100000)
    parser.add_argument('--td3_batch_size', type=int, default=64)
    parser.add_argument('--policy_noise', type=float, default=0.1)
    parser.add_argument('--noise_clip', type=float, default=0.25)
    parser.add_argument('--explore_noise', type=float, default=0.1)

    # Evaluate config
    parser.add_argument('--eval_n_episodes', type=int, default=100)
    parser.add_argument('--load_time_str', type=str, default='')
    parser.add_argument('--eval_env_size', type=ast.literal_eval, default=[40.0, 40.0])
    parser.add_argument('--eval_save_frame', type=bool, default=True)
    parser.add_argument('--eval_terrain_specific_name', type=str, default='white')
    parser.add_argument('--eval_train_type', type=str, default='feature_combine_weight')
    parser.add_argument('--eval_reward_weight_collision', type=float, default=1.0)
    parser.add_argument('--eval_reward_weight_terrain', type=float, default=1.0)

    # Log and painting
    parser.add_argument('--log_update_freq', type=int, default=100)

    args = parser.parse_args()
    args.device = 'cuda:0' if torch.cuda.is_available() else 'cpu'
    return args


def save_args_to_file(args, path, file_name):
    """
    将 args 中的参数信息保存到文本文件。

    :param args: 包含参数的对象，通常是 argparse.Namespace 的实例。
    :param file_path: 要保存文件的路径。
    """
    if not os.path.exists(path):
        os.mkdir(path)

    file_path = os.path.join(path, file_name)
    with open(file_path, 'w') as file:
        for arg, value in vars(args).items():
            file.write(f'{arg}: {value}\n')


if __name__ == '__main__':
    print(parse_args())
